LanguageFeatures/Augmentation-libraries/augmenting_constructors_A04_t01_lib.dart

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion It is a compile-time error if:
+/// ...
+/// - The introductory constructor is marked `factory` and the augmenting
+/// constructor is not, or vice versa.
+///
+/// @description Checks that it is a compile-time error if the introductory
+/// constructor is marked `factory` and the augmenting constructor is not.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmenting_constructors_A04_t01.dart';
+
+augment class C {
+  augment C.foo();
```

Comment:
Do we have a test showing that it is _not_ an error for this constructor to be minimal (e.g., because we just want to add metadata). For example, given that `factory C.foo() = C;` can be augmented by `@metadata augment factory C.foo() = C;`, can we also use `@metadata augment factory C.foo();`?

That would require a grammar change, in line with https://github.com/dart-lang/language/issues/4025.

Let's just stick to the feature specification for now, and update tests if 4025 causes changes to the spec.

Reply:
No, we don't have such test yet. Keeping my eye on 4025.

Reply:
Good, thanks!